bin/docker-image-tool.sh

```diff
@@ -79,7 +79,7 @@ function build {
   fi
 
   # Verify that Spark has actually been built/is a runnable distribution
-  # i.e. the Spark JARs that the Docker files will place into the image are present
```

Comment:
Hi, @rvesse .
Do you have any clue about the special characters after '#' found by @kiszk at [here](https://github.com/apache/spark/pull/22782#issuecomment-431599457)?
I'm wondering if we can avoid this situation in the future. Do you have any idea?

Reply:
Right. We need some automatic check for scripts too. Anyway, thank you for fixing this, @HyukjinKwon ! This blocks almost everything PRs.

Reply:
@dongjoon-hyun This was edited in Xcode on OS X so almost certainly defaulting to UTF-8 encoding

Reply:
It was more because non-breakable space was used instead instead of space. It was in utf-8 but non ascii compatible.

Reply:
Obvious question but why are we still using ASCII encoding for anything?

Reply:
For the issue itself, It's related to a historical reason for Python. Python 2 supported `str` type as bytes like string. It looked a mistake that confuses users about the concept between bytes and string, and then Python 3 introduced `str` as unicode strings concepts like other programing languages.

`open(...).read()` reads it as `str` (which is bytes) in Python 2 but it's read in unicode strings in Python 3 - where we need an implicit conversion between bytes and strings. Looks it had to be to minimise the breaking changes in users codes.

So, bytes to string conversion happened here and unfortunately our Jenkins's system default encoding is set to ascii (even though arguably UTF-8 is common).


For non-ascii itself, please see the justification at http://www.scalastyle.org/rules-dev.html in ScalaStyle.

Reply:
Therefore, using non-breakable spaces in the codes is obviously not a good practice. Please avoid next time.

Reply:
It wasn't as if I used a non-breaking space intentionally, I just used my OSes default editor for Shell scripts!